cpp/src/io/parquet/reader_impl.cpp

```diff
@@ -557,13 +559,17 @@ table_with_metadata reader::impl::finalize_output(
     _output_metadata = std::make_unique<table_metadata>(out_metadata);
   }
 
-  // advance output chunk/subpass/pass info
-  if (_file_itm_data.num_passes() > 0) {
+  // advance output chunk/subpass/pass info for non-empty tables if and only if we are in bounds
+  if (_file_itm_data._current_input_pass < _file_itm_data.num_passes()) {
     auto& pass    = *_pass_itm_data;
     auto& subpass = *pass.subpass;
     subpass.current_output_chunk++;
     _file_itm_data._output_chunk_count++;
```

Comment:
> That said, I wonder if we can derive if we're in the first pass from the existing state.

We can certainly determine this by taking `_file_itm_data._output_chunk_count++` out of the if clause but it might not be segfault safe if somebody uses this without verifying `_file_itm_data._output_chunk_count < _file_itm_data.num_passes()`